tests/testthat/test-app_pws.R

```diff
@@ -0,0 +1,300 @@
+
+  # testing clicking / hovering / brushing
+  test_hover <- shiny::isolate(vals$export$plot_data$hover())
+  testthat::expect_equal(
+    test_hover,
+    hover_vals
+  )
+
+  test_dblclick <- shiny::isolate(vals$export$plot_data$dblclick())
+  testthat::expect_equal(
+    test_dblclick,
+    dbl_click_vals
+  )
+
+  test_click <- shiny::isolate(vals$export$plot_data$click())
+  testthat::expect_equal(
+    test_click,
+    click_vals
+  )
+
+  test_brush <- shiny::isolate(vals$export$plot_data$brush())
+  testthat::expect_equal(
+    test_brush,
+    brush_vals
+  )
+
+  # reset brush to character(0) for next tests
+  app$set_inputs(
+    `plot_with_settings-plot_brush` = character(0),
+    allow_no_input_binding_ = TRUE
+  )
+})
+
+# test output that is returned (reactives and graphic encoded in base64)
+testthat::test_that("plot_with_settings: output is returned", {
+  skip_on_cran()
+  skip_on_ci()
+
+  vals <- app$get_values()
+
+  # check if outputs are reactive
+  for (react_i in vals$export$"plot_data") {
+    testthat::expect_true(is(react_i, "reactive"))
+  }
+
+  # check if output is desired plot
+  testthat::expect_type(
+    vals$output$`plot_with_settings-plot_main`$src, "character"
+  )
+  # nolint start
+  testthat::expect_true(
+    grepl("data:image/png;base64,",
+      vals$output$`plot_with_settings-plot_main`$src,
+      fixed = TRUE
+    )
+  )
+  # nolint end
+})
+
+# download plots. expect_download() might not be stable, hence we test
+# setting inputs and plot name changes
+testthat::test_that("plot_with_settings: download functionality ggplot2", {
+  skip_on_cran()
+  skip_on_ci()
+
+  # test default download options
+  app$click("plot_with_settings-downbutton-downl")
+  date <- strftime(Sys.time(), format = "%Y%m%d")
+  testthat::expect_true(
+    grepl(
+      paste0("plot_", date),
+      app$get_value(input = "plot_with_settings-downbutton-file_name")
+    )
+  )
+  testthat::expect_equal(
+    app$get_value(input = "plot_with_settings-downbutton-file_format"), "png"
+  )
+
+  # Download svg format
+  app$set_inputs(`plot_with_settings-downbutton-downl_state` = TRUE)
+  app$set_inputs(`plot_with_settings-downbutton-file_format` = "svg")
+  app$set_inputs(`plot_with_settings-downbutton-file_name` = "plot_svg")
+  app$click("plot_with_settings-downbutton-downl")
+  testthat::expect_equal(
+    app$get_value(input = "plot_with_settings-downbutton-file_name"), "plot_svg"
+  )
+
+  # change plot width and height
+  app$click("plot_with_settings-expbut")
+  app$set_inputs(`plot_with_settings-width` = 300)
+  app$set_inputs(`plot_with_settings-height` = 300)
+  app$click("plot_with_settings-downbutton-downl")
+  vals <- app$get_values()
+
+  testthat::expect_equal(app$get_value(input = "plot_with_settings-downbutton-file_name"), "plot_svg")
+  testthat::expect_equal(isolate(vals$output$`plot_with_settings-plot_main`$height), 300)
+  testthat::expect_equal(isolate(vals$output$`plot_with_settings-plot_main`$width), 300)
```

Comment:
This test doesn't make sense to me. Values of the inputs are set and then they are checked whether they were checked. In this test we actually should check whether file has been downloaded if the file has expected name, width and height